client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/hooks/use-recipe.ts

```diff
@@ -0,0 +1,197 @@
+import { useFontPairingVariations } from '@automattic/global-styles';
+import { useViewportMatch } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { ONBOARD_STORE } from '../../../../../stores';
+import type { GlobalStyles, OnboardSelect, StarterDesigns } from '@automattic/data-stores';
+import type { Design, StyleVariation } from '@automattic/design-picker';
+import type { GlobalStylesObject } from '@automattic/global-styles';
+
+const useRecipe = (
+	siteId = 0,
+	allDesigns: StarterDesigns | undefined,
+	pickDesign: ( design?: Design ) => void,
+	recordPreviewDesign: ( design: Design, styleVariation?: StyleVariation ) => void,
+	recordPreviewStyleVariation: ( design: Design, styleVariation?: StyleVariation ) => void
+) => {
+	const [ searchParams, setSearchParams ] = useSearchParams();
+	const isDesktop = useViewportMatch( 'large' );
+	const [ isPreviewingDesign, setIsPreviewingDesign ] = useState( false );
+	const { selectedDesign, selectedStyleVariation } = useSelect( ( select ) => {
+		const { getSelectedDesign, getSelectedStyleVariation } = select(
+			ONBOARD_STORE
+		) as OnboardSelect;
+		return {
+			selectedDesign: getSelectedDesign(),
+			selectedStyleVariation: getSelectedStyleVariation(),
+		};
+	}, [] );
+
+	const { setSelectedDesign, setSelectedStyleVariation } = useDispatch( ONBOARD_STORE );
+	const [ selectedFontVariation, setSelectedFontVariation ] = useState< GlobalStylesObject | null >(
+		null
+	);
+	const [ globalStyles, setGlobalStyles ] = useState< GlobalStylesObject | null >( null );
+	const preselectedTheme = searchParams.get( 'theme' );
+	const preselectedStyle = searchParams.get( 'style_variation' );
+	const preselectedFontVariationTitle = searchParams.get( 'font_variation_title' );
+
+	const { stylesheet = '' } = selectedDesign?.recipe || {};
+
+	const fontVariations = useFontPairingVariations(
+		siteId,
+		stylesheet,
+		! preselectedFontVariationTitle
+	);
+
+	const handleSelectedDesignChange = ( design?: Design ) => {
+		setSelectedDesign( design );
+		setSearchParams( ( currentSearchParams ) => {
+			if ( design && design.is_virtual && design.recipe?.slug ) {
+				currentSearchParams.set( 'theme', design.recipe?.slug );
+				currentSearchParams.set( 'is_virtual', 'true' );
```

Comment:
Where are we using / consuming this `is_virtual` query param?

Reply:
I wanted to use it to determine which slug we have to use but finally I simply checked both as followed:
```ts
design.recipe?.slug === preselectedTheme || design.slug === preselectedTheme
```

I'll remove it since it's no longer used